src/engine/runtime.js

```diff
@@ -699,10 +705,11 @@ class Runtime extends EventEmitter {
             this._refreshTargets = false;
         }
 
-        // @todo(vm#570) only emit if monitors has changed since last time.
-        this.emit(Runtime.MONITORS_UPDATE,
-            Object.keys(this._monitorState).map(key => this._monitorState[key])
-        );
+        if (!this._prevMonitorState.equals(this._monitorState)) {
+            const monitorStateObj = this._monitorState.toJS();
+            this.emit(Runtime.MONITORS_UPDATE, Object.keys(monitorStateObj).map(key => monitorStateObj[key]));
```

Comment:
I've changed it and added https://github.com/LLK/scratch-gui/pull/381

It works, but it gives me a weird warning in the console:
Warning: Failed prop type: Invalid prop `monitors` of type `Map` supplied to `MonitorList`, expected instance of `Map`.
    in MonitorList (created by MonitorList)

Also I had to convert to JS in the GUI since it gave me a warning that having Maps as UI elements was a temporary feature that might go away.